app/src/androidTest/java/com/infomaniak/drive/ApiRepositoryTest.kt

```diff
@@ -0,0 +1,521 @@
+        Assert.assertNotNull("test folder must not be null", folder)
+        val postResponse = postFolderAccess(folder!!)
+        assertApiResponse(postResponse)
+        deleteTestFile(folder)
+    }
+
+    @Test
+    fun manageDropboxLifecycle() {
+        // Create a folder to convert it in dropbox
+        val name = "testFold"
+        val folder = createFolder(okHttpClient, userDrive.driveId, ROOT_ID, name, false).data
+        Assert.assertNotNull(folder)
+        // No dropbox yet
+        assertApiResponse(getDropBox(folder!!), false)
+
+        val maxSize = 16384L
+        val body = arrayMapOf(
+            "email_when_finished" to true,
+            "limit_file_size" to maxSize,
+            "password" to "password"
+        )
+        // Add a dropBox
+        val dropboxId = postDropBox(folder, body).let {
+            assertApiResponse(it)
+            Assert.assertTrue("Email when finished must be true", it.data!!.emailWhenFinished)
+            Assert.assertEquals("Limit file size should be $maxSize", maxSize, it.data!!.limitFileSize)
+            it.data!!.id
+        }
+
+        getDropBox(folder).also {
+            assertApiResponse(it)
+            Assert.assertEquals("Dropbox name should be '$name'", name, it.data!!.alias)
+            Assert.assertEquals("Dropbox id should be $dropboxId", dropboxId, it.data!!.id)
+        }
+        val data = JsonObject().apply {
+            addProperty("email_when_finished", false)
+            addProperty("limit_file_size", maxSize * 2)
+        }
+
+        updateDropBox(folder, data).also {
+            assertApiResponse(it)
+            Assert.assertTrue(it.data!!)
+        }
+
+        getDropBox(folder).also {
+            assertApiResponse(it)
+            Assert.assertEquals("Dropbox id should be $dropboxId", dropboxId, it.data!!.id)
+            Assert.assertEquals("Email when finished should be false", false, it.data!!.emailWhenFinished)
+            Assert.assertEquals("Limit file size should be ${maxSize * 2}", maxSize * 2, it.data!!.limitFileSize)
+        }
+
+        assertApiResponse(deleteDropBox(folder))
+        // No dropbox left
+        assertApiResponse(getDropBox(folder), false)
+        deleteTestFile(folder)
+    }
+
+    @Test
+    fun manageTrashLifecycle() {
+        createFileForTest().also { file ->
+            val newName = "Trash test"
+            renameFile(file, newName)
+            val modifiedFile = ApiRepository.getLastModifiedFiles(userDrive.driveId).data?.first()
+            Assert.assertNotNull(modifiedFile)
+            deleteTestFile(modifiedFile!!)
+            getTrashFile(modifiedFile, File.SortType.RECENT).also {
+                assertApiResponse(it)
+                Assert.assertEquals("file id should be the same", it.data?.id, file.id)
+                Assert.assertEquals("file name should be updated to 'Trash test'", newName, it.data?.name)
```

Comment:
```suggestion
                it.data?.run {
                    Assert.assertEquals("file id should be the same", id, file.id)
                    Assert.assertEquals("file name should be updated to 'Trash test'", newName, name)
                }
```

Reply:
If data is null, the test will not be run, so don't do this